src/pages/ImportData.js

```diff
@@ -119,9 +136,35 @@ const ImportData = () => {
       }],
       ['snapshot', async () => {
         snapshots && snapshots.length > 0 ?
-          await Promise.allSettled(_.map(({ title, id }) => Ajax().Workspaces.workspace(namespace, name).importSnapshot(id, title), snapshots)) :
-          await Ajax().Workspaces.workspace(namespace, name).importSnapshot(snapshotId, snapshotName)
-        notify('success', 'Snapshot imported successfully.', { timeout: 3000 })
+          await Promise.allSettled(
+            _.map(({ title, id, description }) => {
+              // Normalize the title:
+              // Importing snapshot will throw an "enum" error if the title has any spaces or special characters
+              // Replace all whitespace characters with _
+              // Then replace all non alphanumeric characters with nothing
+              const normalizedTitle = title.replace(/\s/g, '_').replace(/[^A-Za-z0-9\s-_]/g, '')
+              return Ajax().Workspaces.workspace(namespace, name).importSnapshot(id, normalizedTitle, description)
+            }, snapshots)
+          ).then(async responses => {
+            if (_.some(({ status }) => status === 'rejected', responses)) {
+              const normalizedResponses = await Promise.all(_.map(async ({ status, reason }) => {
+                const reasonJson = reason ? await reason.json() : '{}'
+                const message = JSON.parse(reasonJson.message || '{}').message
+                return { status, message }
+              }, responses))
+              setSnapshotResponses(normalizedResponses)
+
+              // Consolidate the multiple errors into a single error message
+              throw new Error(_.flow(
+                _.filter(({ status }) => status === 'rejected'),
+                _.map('message'),
+                _.join(', ')
+              )(normalizedResponses))
+            }
+          }) :
+          await Ajax().Workspaces.workspace(namespace, name).importSnapshot(snapshotId, snapshotName).then(() => {
+            notify('success', 'Snapshot imported successfully.', { timeout: 3000 })
+          })
```

Comment:
The long ternary here can be hard to follow, along with the `.then` chain. I think `await` cleans this up a bit. Also, since we are displaying the error message beneath each entry, I think we can keep the error notification short.

```suggestion
        if (snapshots?.length > 0) {
          const responses = await Promise.allSettled(
            _.map(({ title, id, description }) => {
              // Normalize the title:
              // Importing snapshot will throw an "enum" error if the title has any spaces or special characters
              const normalizedTitle = title.replace(/\s/g, '_').replace(/[^A-Za-z0-9\s-_]/g, '')
              return Ajax().Workspaces.workspace(namespace, name).importSnapshot(id, normalizedTitle, description)
            }, snapshots)
          )

          if (_.some({ status: 'rejected' }, responses)) {
            const normalizedResponses = await Promise.all(_.map(async ({ status, reason }) => {
              const reasonJson = reason ? await reason.json() : '{}'
              const message = JSON.parse(reasonJson.message || '{}').message
              return { status, message }
            }, responses))

            setSnapshotResponses(normalizedResponses)

            const numFailures = _.size(_.filter(({ status }) => status === 'rejected', normalizedResponses))
            throw new Error(`${numFailures} workspace${numFailures > 1 ? 's' : ''} failed to import. ` +
              'See details in the "Linking to Workspace" section')
          }
        } else {
          await Ajax().Workspaces.workspace(namespace, name).importSnapshot(snapshotId, snapshotName).then(() => {
            notify('success', 'Snapshot imported successfully.', { timeout: 3000 })
          })
        }
```

Reply:
It's really hard for me to see the difference between this code and the one I have - Did you just mkae one change to replace the ternary with an if/else?

Reply:
Sorry - the diff is too big. 
1. Change to the ternary
2. Removed the `then` in favor of capturing the responses in the `await`
3. Changed the error notification message

Reply:
ok updated 👍